crates/sui-framework/packages/sui-framework/sources/coin_metadata_registry.move

```diff
@@ -0,0 +1,208 @@
+module sui::coin_metadata_registry;
+
+use std::string::String;
+use sui::balance::Supply;
+use sui::coin::{TreasuryCap, CoinMetadata, RegulatedCoinMetadata};
+use sui::dynamic_field;
+use sui::transfer::Receiving;
+
+const ENotSystemAddress: u64 = 0;
+const EMetadataAlreadyExists: u64 = 1;
+const EMetadataNotFound: u64 = 2;
+const EAlreadyClaimed: u64 = 3;
+const EMetadataCapNotClaimed: u64 = 4;
+const EAlreadyRegulated: u64 = 5;
+
+public struct Registry has key { id: UID }
+
+// === Coin Metadata ===
+public struct CoinMetadataKey<phantom T>() has copy, drop, store;
+
+public struct CoinMetadataCap<phantom T> has key, store { id: UID }
+
+public struct MetadataCapClaimed<phantom T>() has copy, drop, store;
+
+public struct Metadata<phantom T> has store {
+    decimals: u8,
+    name: String,
+    symbol: String,
+    description: String,
+    icon_url: String,
+    is_regulated: Option<bool>,
```

Comment:
:fire:

Reply:
I would ask around if there are any other fields we want to put here, maybe there are extra parameters for it.